thriftbp/server.go

```diff
@@ -97,30 +97,92 @@ func NewBaseplateServer(
 	bp baseplate.Baseplate,
 	cfg ServerConfig,
 ) (baseplate.Server, error) {
-	middlewares := BaseplateDefaultProcessorMiddlewares(
-		DefaultProcessorMiddlewaresArgs{
-			EdgeContextImpl:                    bp.EdgeContextImpl(),
-			ErrorSpanSuppressor:                cfg.ErrorSpanSuppressor,
-			ReportPayloadSizeMetricsSampleRate: cfg.ReportPayloadSizeMetricsSampleRate,
+
+	opts := []ServerOpt{
+		func(optCfg ServerConfig) ServerConfig {
+			return cfg
 		},
```

Comment:
These lines are the reason why I opted for having the `ServerOpt` return the `ServerConfig` instance instead of unit. It was the most straight forward way I could think of maintaining backwards compatibility. Would using a pointer as an arg be the better option?